Include/dpsim/DataLogger.h

```diff
@@ -58,22 +58,10 @@ namespace DPsim {
 
 		void setColumnNames(std::vector<String> names);
 
-		void addAttribute(const String &name, CPS::AttributeBase::Ptr attr);
-		void addAttribute(const String &name, CPS::Attribute<Int>::Ptr attr);
-		void addAttribute(const String &name, CPS::Attribute<Real>::Ptr attr);
-		void addAttribute(const String &name, CPS::Attribute<Complex>::Ptr attr);
-		void addAttribute(const String &name, CPS::MatrixRealAttribute::Ptr attr);
-		void addAttribute(const String &name, CPS::MatrixCompAttribute::Ptr attr, UInt rowsMax = 0, UInt colsMax = 0);
-		void addAttribute(const String &name, const String &attr, CPS::IdentifiedObject::Ptr obj);
-		void addAttribute(const String &name, const String &attr, CPS::IdentifiedObject::Ptr obj, UInt rowsMax, UInt colsMax);
-		void addAttribute(const std::vector<String> &name, const String &attr, CPS::IdentifiedObject::Ptr obj);
-		void addAttribute(const std::vector<String> &name, CPS::AttributeBase::Ptr attr);
-		void addAttribute(const std::vector<String> &name, CPS::MatrixRealAttribute::Ptr attr);
-
-		template<typename VarType>
-		void addNode(typename CPS::SimNode<VarType>::Ptr node) {
-			addAttribute(node->name() + ".voltage", node->attributeMatrix("voltage"));
-		}
+		void logAttribute(const String &name, CPS::AttributeBase::Ptr attr, UInt rowsMax = 0, UInt colsMax = 0);
+
+		///DEPRECATED: Only use for compatiblity, otherwise this just adds extra overhead to the logger. Instead just call logAttribute multiple times for every coefficient.
```

Comment:
The comment relates to the function `void logAttribute(const std::vector<String> &name, CPS::AttributeBase::Ptr attr);` which takes a (Matrix-)Attribute and a list of strings for the log names. This function is used in very few places and each time, the list of strings has to be generated before, so one could just as easily call the first logger function for every coefficient which should receive a name. Single coefficients can be logged by calling the first `logAttribute` function with `attr->deriveCoeff<>(a,b)`. The `rowsMax` and `colsMax` parameters are not relevant for this usecase. I will clarify the use of the derive-function in the comment.

Reply:
Optimally, we would update the logger to not use the `derive`-functions at all, but instead keep a list of coefficients and names to log for every matrix as this would probably improve performance.